store/src/java/com/zimbra/cs/imap/ImapSessionManager.java

```diff
@@ -171,9 +182,12 @@ public void run() {
 
                 // As we are more likely to decide to drop or page out sessions we process later, we want to start
                 // with the most recently used ones as they are more likely to be useful again.
-                Iterator<ImapListener> mostRecentToLeastRecentIterator = sessions.descendingKeySet().iterator();
+                Iterator<String> mostRecentToLeastRecentIterator = sessions.descendingKeySet().iterator();
                 while (mostRecentToLeastRecentIterator.hasNext()) {
-                    ImapListener session = mostRecentToLeastRecentIterator.next();
+                    ImapListener session =  sessions.get(mostRecentToLeastRecentIterator.next());
+                    ZimbraLog.imap.debug("Session: %s interactive? %s nonInteractiveCutoff=%d cutoff=%d lastAccessed=%d",
```

Comment:
May not be a new issue but I would be happier if we had code handling session being null here.

Reply:
Handled.